frontend/reducers/basicAuth.js

```diff
@@ -0,0 +1,59 @@
+import {
+  basicAuthFetchStartedType as FETCH_STARTED,
+  basicAuthReceivedType as RECEIVED,
+  basicAuthSavedType as SAVED,
+  basicAuthRemovedType as REMOVED,
+} from '../actions/actionCreators/basicAuthActions';
+
+import {
+  httpErrorType as HTTP_ERROR,
+} from '../actions/actionCreators/alertActions';
+
+const initialState = {
+  isLoading: false,
+  data: {},
+};
+
+export default function basicAuth(state = initialState, action) {
+  switch (action.type) {
+    case HTTP_ERROR: {
+      return {
+        isLoading: false,
+        data: {},
+      };
+    }
+
+    case FETCH_STARTED: {
+      return {
+        isLoading: true,
+        data: {},
+      };
+    }
+
+    case RECEIVED: {
+      const { basicAuth: credentials } = action.payload;
+      return {
+        isLoading: false,
+        data: credentials,
+      };
+    }
+
+    case SAVED: {
+      const { basicAuth: credentials } = action.payload;
+      return {
+        isLoading: false,
+        data: credentials,
+      };
+    }
+
+    case REMOVED: {
+      return {
+        isLoading: false,
+        data: {},
+      };
+    }
+
+    default:
+      return state;
```

Comment:
For the UEVs, the state is keyed by site Id, but these aren't, is there a reason why they should be treated differently? This implies that the state contains the basic auth credentials for the *current* site, but this is a global store and could lead to problems if the site changed but the BasicAuth component(s) was/were NOT remounted.

That said, since this data is part of the site schema, why not just include it when fetching the site?